projects/valtimo/logging/src/lib/components/log-search/log-search.component.ts

```diff
@@ -0,0 +1,241 @@
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {CommonModule} from '@angular/common';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import {FormArray, FormBuilder, ReactiveFormsModule} from '@angular/forms';
+import {TrashCan16} from '@carbon/icons';
+import {TranslateModule} from '@ngx-translate/core';
+import {CARBON_THEME, CdsThemeService, CurrentCarbonTheme} from '@valtimo/components';
+import {
+  ButtonModule,
+  DatePicker,
+  DatePickerModule,
+  DropdownModule,
+  IconModule,
+  IconService,
+  InputModule,
+  ListItem,
+} from 'carbon-components-angular';
+import flatpickr from 'flatpickr';
+import {debounceTime, map, Observable, Subscription} from 'rxjs';
+import {
+  LoggingEventProperty,
+  LoggingEventSearchFormValue,
+  LoggingEventSearchRequest,
+  LogLevel,
+} from '../../models';
+
+@Component({
+  selector: 'valtimo-log-search',
+  templateUrl: './log-search.component.html',
+  styleUrl: './log-search.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+  imports: [
+    CommonModule,
+    TranslateModule,
+    ButtonModule,
+    DropdownModule,
+    IconModule,
+    InputModule,
+    ReactiveFormsModule,
+    DatePickerModule,
+  ],
+})
+export class LogSearchComponent implements OnInit, AfterViewInit, OnDestroy {
+  @ViewChild('afterTimestamp') private readonly _afterTimestampDatePicker: DatePicker;
+  @ViewChild('beforeTimestamp') private readonly _beforeTimestampDatePicker: DatePicker;
+
+  @Input() public set initSearchRequest(value: LoggingEventSearchRequest) {
+    const mappedFormValue: LoggingEventSearchFormValue = this.mapSearchRequestToFormValue(value);
+    this.initLogItems(mappedFormValue);
+    this.initPropertiesForm(mappedFormValue);
+
+    this.formGroup.patchValue(mappedFormValue, {emitEvent: false});
+  }
+  @Output() public readonly searchSubmitEvent = new EventEmitter<LoggingEventSearchRequest>();
+
+  public readonly theme$: Observable<CARBON_THEME> = this.cdsThemeService.currentTheme$.pipe(
+    map((theme: CurrentCarbonTheme) =>
+      theme === CurrentCarbonTheme.G10 ? CARBON_THEME.WHITE : CARBON_THEME.G100
+    )
+  );
+
+  public readonly formGroup = this.fb.group({
+    likeFormattedMessage: this.fb.control<string>(''),
+    level: this.fb.control<ListItem>({content: '', selected: false}),
+    beforeTimestamp: this.fb.control<string>(''),
+    afterTimestamp: this.fb.control<string>(''),
+    properties: this.fb.array([
+      this.fb.group({
+        key: this.fb.control<string>(''),
+        value: this.fb.control<string>(''),
+      }),
+    ]),
+  });
+
+  public logLevelItems: ListItem[] = [
```

Comment:
better make this Partial<ListItem> so you don't have to do "selected: false" for each one, you're mapping this later anyway